src/report.sh

```diff
@@ -350,9 +351,13 @@ function save_data_to()
     exit "$ret"
   fi
 
-  truncate -s 0 "$path"
-  [[ -n "${options_values['POMODORO']}" ]] && show_data "$target_time" >> "$path"
-  [[ -n "${options_values['STATISTICS']}" ]] && show_statistics >> "$path"
+  if [[ -d $path ]]; then
+    [[ -n "${options_values['POMODORO']}" ]] && show_data "$target_time" >> "$path/$output"
+    [[ -n "${options_values['STATISTICS']}" ]] && show_statistics >> "$path/$output"
+  else
```

Comment:
I think the best choice here is to use the `-w` check.

Reply:
You still didn't add these tests, I think it's unlikely that an invalid path would get to this point as it would probably fail at the touch command, still, it's best to be sure.